webboot/webboot.go

```diff
@@ -212,13 +220,50 @@ func main() {
 			log.Fatal(err)
 		}
 	}
+
+
+
+
+	ifRE := regexp.MustCompilePOSIX(*ifname)
+
+
+	ifnames, err := netlink.LinkList()
+	if err != nil {
+		log.Fatalf("Can't get list of link names: %v", err)
+	}
+
+	var filteredIfs []netlink.Link
+	for _, iface := range ifnames {
+		if ifRE.MatchString(iface.Attrs().Name) {
+			filteredIfs = append(filteredIfs, iface)
+		}
+	}
+
+
+
+
+	packetTimeout := time.Duration(*timeout) * time.Second
+
+	ctx, cancel := context.WithTimeout(context.Background(), packetTimeout*time.Duration(1<<uint(*retry)))
+	defer cancel()
+
+	c := dhclient.Config{
+		Timeout: packetTimeout,
+		Retries: *retry,
+	}
+	if *verbose {
+		c.LogLevel = dhclient.LogSummary
+	}
+
 	//the Request function sets up a DHCP confifuration for all interfaces,
 	//such as eth0, which is an ethernet interface.
 	if *ipv4 || *ipv6 {
 		//ifname uses the regular expression ^[we].* to check for an interface starting with w or e such as
 		//wlan0/1, enx453243, or eth0/1
-		dhclient.Request(*ifName, *timeout, *retry, *verbose, *ipv4, *ipv6)
 
+		//dhclient.Request(*ifName, *timeout, *retry, *verbose, *ipv4, *ipv6)
+		dhclient.IfUp(*ifname)
+		dhclient.SendRequests(ctx, filteredIfs, *ipv4, *ipv6, c)
```

Comment:
you are ignoring the return value of sendrequests. You should wait for at least one successful return value, so that you know at least one interface is up and ready. Failure to do this is what panics my kernels.

Reply:
will do.